library/src/blas1/rocblas_asum.hpp

```diff
@@ -28,18 +25,14 @@ rocblas_status rocblas_asum_template(rocblas_handle handle,
                                      To*            workspace,
                                      To*            result)
 {
```

Comment:
We aren't calling these templates anymore are we? Are we planning on keeping them around only for "internal API" purposes?

Reply:
Good point. 
We could keep for internal purpose :) What do you think?

Reply:
I would vote "yes" I think. We don't need them for anything yet afaik, but we might in the future, and its good for consistency. We could potentially just use the `reduction_template` if we ever need to, but I think that just introduces some confusion. I could be easily swayed, though, since removing them would get rid of a decent chunk of code.

Reply:
I tried to stay consistent with the C-API === IMPL === TEMPLATE

Reply:
Yeah, sounds good to me. In that case we don't need the batched and strided_batched templates, and the regular templates, `rocblas_asum_template` for example, should be generic enough to handle batched/ strided_batched cases.

Reply:
that's a possibility ! :) let's see what other REVIEWERS! have to say :p

Reply:
If we want to export compiled templates then there is some value in making the explicit templates that have a clear mapping to functionality.